webapp/channels/src/components/advanced_text_editor/advanced_text_editor.test.tsx

```diff
@@ -0,0 +1,326 @@
+
+import type {PostDraft} from 'types/store/draft';
+
+import AdavancedTextEditor from './advanced_text_editor';
+
+global.ResizeObserver = require('resize-observer-polyfill');
+
+const currentUserId = 'current_user_id';
+const channelId = 'current_channel_id';
+
+const initialState = {
+    entities: {
+        general: {
+            config: {
+                EnableConfirmNotificationsToChannel: 'false',
+                EnableCustomGroups: 'false',
+                PostPriority: 'false',
+                ExperimentalTimezone: 'false',
+                EnableCustomEmoji: 'false',
+                AllowSyncedDrafts: 'false',
+            },
+            license: {
+                IsLicensed: 'false',
+                LDAPGroups: 'false',
+            },
+        },
+        channels: {
+            channels: {
+                current_channel_id: {
+                    id: 'current_channel_id',
+                    group_constrained: false,
+                    team_id: 'current_team_id',
+                    type: 'O',
+                },
+            },
+            stats: {
+                current_channel_id: {
+                    member_count: 1,
+                },
+            },
+            roles: {
+                current_channel_id: new Set(['channel_roles']),
+            },
+            groupsAssociatedToChannel: {},
+            channelMemberCountsByGroup: {
+                current_channel_id: {},
+            },
+        },
+        teams: {
+            currentTeamId: 'current_team_id',
+            teams: {
+                current_team_id: {
+                    id: 'current_team_id',
+                    group_constrained: false,
+                },
+            },
+            myMembers: {
+                current_team_id: {
+                    roles: 'team_roles',
+                },
+            },
+            groupsAssociatedToTeam: {},
+        },
+        users: {
+            currentUserId: 'current_user_id',
+            profiles: {
+                current_user_id: {
+                    id: 'current_user_id',
+                    roles: 'user_roles',
+                    locale: 'en',
+                },
+            },
+            statuses: {
+                current_user_id: 'online',
+            },
+        },
+        roles: {
+            roles: {
+                user_roles: {permissions: []},
+                channel_roles: {permissions: []},
+                team_roles: {permissions: []},
+            },
+        },
+        groups: {
+            groups: {},
+        },
+        emojis: {
+            customEmoji: {},
+        },
+        preferences: {
+            myPreferences: {},
+        },
+    },
+    websocket: {
+        connectionId: 'connection_id',
+    },
+};
+
+const emptyDraft: PostDraft = {
+    message: '',
+    uploadsInProgress: [],
+    fileInfos: [],
+    channelId,
+    rootId: '',
+    createAt: 0,
+    updateAt: 0,
+};
+
+const baseProps = {
```

Comment:
I know this is mostly a product of the poor interface of this component, but I do wonder if it would be a good idea to find a way to generate these mocked props automatically at some point. I suppose that'd mean tests aren't always updated when new props are added though

Reply:
This is supposed to be used only here, so it is not so bad that it is this way. That being said, once we separate the `advanced_text_editor` into domain specific components, it will get way cleaner.